### ISSUE TITLE
Bring sharding to all libraries, add note about sharding

### ISSUE THREAD

**Issue by kellysutton** [imgix/imgix-blueprint#1]
All libraries should include sharding as seen in the Ruby library: https://github.com/imgix/imgix-rb#domain-sharded-urls

Sharding is not a panacea to performance issues and should be used carefully. Note should link to: http://perf.fail/post/96104709544/zealous-sharding-hurts-etsy-performance

Sharding is also an anti-pattern for HTTP/2: https://docs.google.com/presentation/d/1r7QXGYOLCh4fcUq0jDdDwKJWNqWK1o4xMtYpKZCJYjM/present?slide=id.g518e3c87f_0_300

**TL;DR:** If you're going to use sharding, only use 2 shards.

**Comment by sherwinski** [comment]
Closing as we decided to deprecate and remove domain sharding from our official libraries a few years back. See our [blog post](https://blog.imgix.com/2019/05/03/deprecating-domain-sharding) discussing this decision in more detail.
